core/client/src/client.rs

```diff
@@ -1529,12 +1529,14 @@ where
 	B: backend::Backend<Block, Blake2Hasher>,
 	Block: BlockT<Hash=H256>,
 {
-	fn best_block_header(&self) -> Result<<Block as BlockT>::Header, ConsensusError> {
+	fn best_block_header_for_authoring(&self)
```

Comment:
I'm not a big fan of this distinction. what does it mean to be "best for authoring" vs. best in general?

Reply:
Me neither, it is only coming from the pragmatic problem, you mentioned, where we'd have two distinct ways (potentially coming to different outcomes) to determine what the `best_block` is based upon whether we finalise or we propose in polkadot. Or did I misunderstand that?

Reply:
yes, that is one practical problem we have. I think we ought to have an abstraction that covers it and a clear definition of what the separation is.